intake_parquet/source.py

```diff
@@ -106,5 +106,43 @@ def _to_dask(self):
         self._load_metadata()
         return self._df
 
+    def to_spatial(self):
+        """Read a data source as a SpatialPointsFrame.
+
+        This requires that the data be structured in a particular, spatially
+        optimized, way. To convert a regular dataframe to a SpatialPointsFrame
+        use the functionality in datshader.
+
+        >>> import datashader.spatial.points as dsp
+        >>> dsp.to_parquet(df, 'sorted.parq', 'x', 'y', shuffle='disk', npartitions=32)
+
+        More info at http://datashader.org/user_guide/2_Points.html
+
+        This whole function is copied from datashader.
+        https://github.com/pyviz/datashader/blob/815596bbf72017c6b8014cfc96354601a3529b04/datashader/spatial/points.py#L297-L312
+        """
+        try:
+            from datashader.spatial.points import SpatialPointsFrame
+        except ImportError:
+            raise ImportError('SpatialPointsFrame not found in this '
+                'version of datashader. Get latest using '
+                '`conda install -c pyviz datashader`.')
+        import json
+        import fastparquet as fp
+
+        frame = self._df or self.to_dask()
+        urlpath =  self._get_cache(self._urlpath)[0]
+        pf = fp.ParquetFile(urlpath)
```

Comment:
I guess this probably doesn't work for remote. Is there a good way to get `key_value_metadata` off the dask version?

Reply:
No there isn't. You need slightly different things to work with FP directly, but it can be done.